src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginItem.cs

```diff
@@ -32,14 +35,17 @@ namespace Microsoft.CodeAnalysis.InheritanceMargin
 
         public InheritanceMarginItem(
             int lineNumber,
+            string? topLevelDisplayText,
             ImmutableArray<TaggedText> displayTexts,
             Glyph glyph,
+            bool isOrdered,
             ImmutableArray<InheritanceTargetItem> targetItems)
         {
             LineNumber = lineNumber;
+            TopLevelDisplayText = topLevelDisplayText;
             DisplayTexts = displayTexts;
             Glyph = glyph;
-            TargetItems = targetItems;
+            TargetItems = isOrdered ? targetItems : targetItems.OrderBy(item => item.DisplayName).ToImmutableArray();
```

Comment:
Seems like a place might be benefited from our offline discussion. (Move all the data processing part to OOP)
I am fine with this approach now, but could you add a comment here to explain this would only be used for showing the global using?